Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2472,10 +2485,26 @@ - (void)takePictureOrVideo
     }];
 }
 
-- (void)chooseFromLibrary
+- (void)chooseFromLibraryAsDocument
+{
+    OWSAssertIsOnMainThread();
+
+    [self chooseFromLibrary:YES];
+}
+
+- (void)chooseFromLibraryAsMedia
+{
+    OWSAssertIsOnMainThread();
+
+    [self chooseFromLibrary:NO];
+}
+
+- (void)chooseFromLibrary:(BOOL)shouldTreatAsDocument
 {
     OWSAssertIsOnMainThread();
 
+    self.isPickingMediaAsDocument = shouldTreatAsDocument;
```

Comment:
Just checking - can we use the method param rather than storing and accessing this property? 

It would be nice to get rid of unnecessary state tracking, and it seems like it's only used in this method.

Reply:
Unfortunately, due to the delegate pattern we need to hang this state somewhere.  I considered hanging it somewhere else, but couldn't find a better place.

Reply:
Ah you're right of course, I misread the diff.